engine/src/execution_graph/logic_controllers/LogicPrimitives.h

```diff
@@ -58,6 +59,8 @@ class BlazingTableView {
 
 	std::vector<std::string> names() const;
 
+	void setNames(const std::vector<std::string> &names) { this->columnNames = names; }
```

Comment:
a view should have no way to set names. It is a view of a table. It is immutable.

Reply:
I am guessing that this done because otherwise theres no way to manage the lifecycle.